modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/SnapshotOrdinalTimeline.scala

```diff
@@ -0,0 +1,37 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.{NonEmptyList, Validated}
+import cats.syntax.all._
+
+import org.tessellation.node.shared.domain.seedlist.Range.{mergeOverlaps, parseInput}
+
+import fs2.data.csv.{CellDecoder, DecoderError}
+import io.estatico.newtype.macros.newtype
+
+object SnapshotOrdinalTimeline {
+
+  @newtype
+  case class SnapshotOrdinalTimeline(values: NonEmptyList[Range])
+
+  implicit val optionalSnapshotOrdinalTimelineCellDecoder: CellDecoder[Option[SnapshotOrdinalTimeline]] =
+    CellDecoder.stringDecoder.emap { s =>
+      fromString(s)
+        .leftMap(err => new DecoderError(err.getMessage))
+        .toEither
+    }
+
+  def fromString(str: String): Validated[Throwable, Option[SnapshotOrdinalTimeline]] =
+    parseInput(str)
+      .map(mergeOverlaps)
+      .andThen(_.toNel match {
+        case Some(value) => value.validNel
+        case None        => InvalidInput(str).invalidNel
+      })
+      .map(SnapshotOrdinalTimeline(_).some)
+      .leftMap(
+        _.map(_.getMessage).toList
+          .mkString("; ")
+      )
+      .leftMap(InvalidInput)
```

Comment:
why you merge everything into a single string, and then return an `InvalidInput`? can't you either return the list of errors or a single string? depending on how you would like to use it later

Reply:
Re-worked to

```scala
  implicit val optionalSnapshotOrdinalTimelineCellDecoder: CellDecoder[Option[SnapshotOrdinalTimeline]] =
    CellDecoder.stringDecoder.emap { s =>
      decode(s)
        .leftMap(
          _.map(_.getMessage).toList
            .mkString("; ")
        )
        .leftMap(new DecoderError(_))
        .toEither
    }

  def decode(input: String): Validated[NonEmptyList[Throwable], Option[SnapshotOrdinalTimeline]] =
    Ranges
      .decode(input)
      .map(mergeOverlaps)
      .andThen(
        _.toNel match {
          case Some(value) => value.validNel
          case None        => InvalidInput(input).invalidNel
        }
      )
      .map(SnapshotOrdinalTimeline(_).some)
```